client/lib/safe-image-url/index.js

```diff
@@ -1,44 +1,48 @@
 /**
  * External Dependencies
  */
-var photon = require( 'photon' ),
-	uri = require( 'url' );
+import photon from 'photon';
+import { parse as parseUrl } from 'url';
 
 /**
- * Internal Dependencies
+ * Pattern matching URLs to be left unmodified: relative URLs, inline images,
+ * or local file URLs.
+ *
+ * @type {Array}
+ */
+const REGEX_EXEMPT_URL = /^(\/(?!\/)|data:image\/[^;]+;|blob:)/;
+
+/**
+ * Pattern matching Automattic-controlled hostnames
+ *
+ * @type {RegExp}
  */
+const REGEXP_A8C_HOST = /^([-a-zA-Z0-9_]+\.)*(gravatar\.com|wordpress\.com|wp\.com|a8c\.com)$/;
```

Comment:
yay for escaping the `.`s

could we add a test that exploits the old `.` hole?